app/scripts/controllers/permissions/index.js

```diff
@@ -49,6 +52,10 @@ export class PermissionsController {
 
   createMiddleware ({ origin, extensionId }) {
 
+    if (!origin || typeof origin !== 'string') {
```

Comment:
We might be able to restate this to better express the desired input:

```suggestion
    if (typeof origin !== 'string' || !origin.length) {
```

`typeof` should be able to work with a missing argument and this might more closely match the error message we're providing below.